cli/pcluster/commands.py

```diff
@@ -36,109 +36,129 @@
-        region_name=config.region,
-        aws_access_key_id=config.aws_access_key_id,
-        aws_secret_access_key=config.aws_secret_access_key,
+    pcluster_config = PclusterConfig(
+        config_file=args.config_file,
+        file_sections=[AWS, GLOBAL, CLUSTER],
+        cluster_label=args.cluster_template,
+        fail_on_file_absence=True,
     )
+    # get CFN parameters, template url and tags from config
+    cluster_section = pcluster_config.get_section("cluster")
+    cfn_template_url = cluster_section.get_param_value("template_url")
+    cfn_tags = cluster_section.get_param_value("tags")
+    cfn_params = pcluster_config.to_cfn()
 
-    # Get the MasterSubnetId and use it to determine AvailabilityZone
-    if "MasterSubnetId" in config.parameters:
-        master_subnet_id = config.parameters["MasterSubnetId"]
-        try:
-            ec2 = utils.boto3_client("ec2", aws_client_config)
-            availability_zone = (
-                ec2.describe_subnets(SubnetIds=[master_subnet_id]).get("Subnets")[0].get("AvailabilityZone")
-            )
-        except ClientError as e:
-            LOGGER.critical(e.response.get("Error").get("Message"))
-            sys.stdout.flush()
-            sys.exit(1)
-        config.parameters["AvailabilityZone"] = availability_zone
+    _check_for_updates(pcluster_config)
 
-    capabilities = ["CAPABILITY_IAM"]
     batch_temporary_bucket = None
     try:
-        cfn = utils.boto3_client("cloudformation", aws_client_config)
-        stack_name = "parallelcluster-" + args.cluster_name
+        cfn = boto3.client("cloudformation")
+        stack_name = utils.get_stack_name(args.cluster_name)
+        pcluster_version = utils.get_installed_version()
 
         # If scheduler is awsbatch create bucket with resources
-        if config.parameters["Scheduler"] == "awsbatch":
+        if cfn_params["Scheduler"] == "awsbatch":
             batch_resources = pkg_resources.resource_filename(__name__, "resources/batch")
-            batch_temporary_bucket = create_bucket_with_batch_resources(
-                stack_name=stack_name, aws_client_config=aws_client_config, resources_dir=batch_resources
+            batch_temporary_bucket = _create_bucket_with_batch_resources(
+                stack_name=stack_name, resources_dir=batch_resources, region=pcluster_config.region
             )
-            config.parameters["ResourcesS3Bucket"] = batch_temporary_bucket
+            cfn_params["ResourcesS3Bucket"] = batch_temporary_bucket
 
         LOGGER.info("Creating stack named: %s", stack_name)
 
-        cfn_params = [{"ParameterKey": key, "ParameterValue": value} for key, value in config.parameters.items()]
-        tags = [{"Key": t, "Value": config.tags[t]} for t in config.tags]
-        tags.append({"Key": "Version", "Value": version()})
-
+        # Determine the CloudFormation Template URL to use
+        # order is 1) CLI arg 2) Config file 3) default for version + region
```

Comment:
Done for template_url and tags, I'd avoid the one for the params since it would be a single line function.

Unfortunately the C901 cannot be removed yet. :(